### PR TITLE
feat(use-combined-state): remove unnecessary state management when controlled state mode

### PR DESCRIPTION
**Pull request by andresin87**
**TASK**: 

### Description, Motivation and Context
reduces the rerenders and the complexity.

### Types of changes

- [x] 🧠 Refactor

https://github.com/chakra-ui/chakra-ui/blob/main/packages/legacy/hooks/src/use-controllable.ts#L34

### CODE REVIEW

File: packages/hooks/use-combined-state/src/useCombinedState.tsx [hunk 1]
```diff
@@ -1,34 +1,38 @@
...
 
 /**
- * This hook must be used when a component have both a controlled and uncontrolled mode.
- * It will take care of updating the state value when controlled mode (prop) is updated.
+ * This hook must be used when a component has both a controlled and uncontrolled mode.
+ * It will take care of updating the state value when a controlled mode (prop) is updated.
  */
 export function useCombinedState<T>(
   controlledValue?: T,
   defaultValue?: T
```
Comment (andresz1) [inline packages/hooks/use-combined-state/src/useCombinedState.tsx:10]: Great job @andresin87 do you think that would be possible to add the `onChange` callback like chakra does ? 🙏🏻 I think that could be really useful

Comment (andresin87) [reply packages/hooks/use-combined-state/src/useCombinedState.tsx:10]: ok. I'll do today

**Review by andresz1 — APPROVED** [review]

File: packages/hooks/use-combined-state/src/useCombinedState.tsx (hunk 1 above)
Comment (andresin87) [reply packages/hooks/use-combined-state/src/useCombinedState.tsx:10]: 3rd argument

File: packages/hooks/use-combined-state/src/useCombinedState.test.tsx [hunk 2]
```diff
@@ -131,4 +131,62 @@ describe('useCombinedState', () => {
...
+  it('value must NOT be updated using its setter when using uncontrolled mode using shouldUpdate flag to false', () => {
+    // Given
+    const args = { value: undefined, defaultValue: 123 } as argsType
+
+    // When
+    const hook = renderHook(() => useCombinedState(...map(args)))
+    let [response, setResponse, isControlledValue, initialValue] = hook.result.current
+
+    // Then
+    expect(typeof response).toBe('number')
```
Comment (Powerplex) [inline packages/hooks/use-combined-state/src/useCombinedState.test.tsx:144]: I think you might not need some assertions like this one :) We have type safety using TS + other checks implicitely covering it:
`expect(response).toBe(123)`

**Review by Powerplex — APPROVED** [review]

File: packages/hooks/use-combined-state/src/useCombinedState.test.tsx (hunk 2 above)
Comment (andresin87) [reply packages/hooks/use-combined-state/src/useCombinedState.test.tsx]: DONE!

File: packages/hooks/use-combined-state/src/useCombinedState.tsx (hunk 1 above)
Comment (andresin87) [reply packages/hooks/use-combined-state/src/useCombinedState.tsx]: DONE!

**Review by kikoruiz — APPROVED** [review]

**Pull request merged**